server/service/ProductService.js

```diff
@@ -51,8 +55,15 @@ class ProductService {
 
   async deleteProduct(id) {
     try {
-      const deletedProducts = await productModel.destroy({ where: { id } });
-      return deletedProducts;
+      const productToBeDeleted = await productModel.findOne({ where: { id } })
+      if (!productToBeDeleted) {
+        throw new BadRequest('Product with this ID does not exist')
```

Comment:
Do not throw just BadRequest, create a ProductNotFoundException which extends BadRequest